src/components/TokenRow/TokenRow.tsx

```diff
@@ -65,8 +65,8 @@ export function TokenRow<C extends FieldValues>({
               value={value}
               disabled={disabled}
               onValueChange={e => {
-                onInputChange && e.value !== value && onInputChange(e.value)
                 onChange(e.value)
+                if (onInputChange && e.value !== value) onInputChange(e.value)
```

Comment:
Exactly, before this order change, `FormContext` was one change behind, because `onInputChange` was running before `onChange` (which is for internal form context). 
I also used the `if` statement instead of logical and operator, because I think it'll improve code readability.